File: Add Code Here/LeetCode_Sols/Easy/101. Symmetric Tree/Solution.java
class Solution {
    public boolean isSymmetric(TreeNode root) {
        if(root==null) return true;
        return isMirror(root.left,root.right);
    }
    
    public boolean isMirror(TreeNode p, TreeNode q) {
        if(p==null && q==null) return true;
        if(p==null || q==null) return false;
        return (p.val==q.val) && isMirror(p.left,q.right) && isMirror(p.right,q.left);
    }
}

File: Add Code Here/LeetCode_Sols/Easy/101. Symmetric Tree/Solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isSymmetric(self, root: TreeNode) -> bool:
        def isMirror(p, q):
            if not p and not q:
                return True
            if not p or not q:
                return False
            return (p.val == q.val) and isMirror(p.left, q.right) and isMirror(p.right, q.left)
        
        return isMirror(root, root)
